// ionic/dist/commands/cordova/compile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const chalk = require("chalk");
const command_1 = require("@ionic/cli-utils/lib/command");
const base_1 = require("./base");
let CompileCommand = class CompileCommand extends base_1.CordovaCommand {
    preRun(inputs, options) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            yield this.preRunChecks();
            if (!inputs[0]) {
                const platform = yield this.env.prompt({
                    type: 'input',
                    name: 'platform',
                    message: `What platform would you like to compile (${['android', 'ios'].map(v => chalk.green(v)).join(', ')}):`
                });
                inputs[0] = platform.trim();
            }
            yield this.checkForPlatformInstallation(inputs[0]);
        });
    }
    run(inputs, options) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const { ConfigXml } = yield Promise.resolve().then(function () { return require('@ionic/cli-utils/lib/cordova/config'); });
            const { filterArgumentsForCordova } = yield Promise.resolve().then(function () { return require('@ionic/cli-utils/lib/cordova/utils'); });
            const conf = yield ConfigXml.load(this.env.project.directory);
            yield conf.resetContentSrc();
            yield conf.save();
            const response = yield this.runCordova(filterArgumentsForCordova(this.metadata, inputs, options));
            this.env.log.msg(response);
        });
    }
};
CompileCommand = tslib_1.__decorate([
    command_1.CommandMetadata({
        name: 'compile',
        type: 'project',
        description: 'Compile native platform code',
        longDescription: `
Like running ${chalk.green('cordova compile')} directly, but provides friendly checks.
  `,
        exampleCommands: [
            'ios',
            'ios --device',
            'android',
        ],
        inputs: [
            {
                name: 'platform',
                description: `The platform to compile (${['android', 'ios'].map(v => chalk.green(v)).join(', ')})`,
            },
        ],
        options: [
            {
                name: 'debug',
                description: 'Create a Cordova debug build',
                type: Boolean,
                intent: 'cordova',
            },
            {
                name: 'release',
                description: 'Create a Cordova release build',
                type: Boolean,
                intent: 'cordova',
            },
            {
                name: 'device',
                description: 'Compile Cordova build to a device',
                type: Boolean,
                intent: 'cordova',
            },
            {
                name: 'emulator',
                description: 'Compile Cordova build to an emulator',
                type: Boolean,
                intent: 'cordova',
            },
        ],
    })
], CompileCommand);
exports.CompileCommand = CompileCommand;
